src/shared/varlink-serialize.c

```diff
@@ -83,3 +84,14 @@ int varlink_server_deserialize_one(sd_varlink_server *s, const char *value, FDSe
         LIST_PREPEND(sockets, s->sockets, TAKE_PTR(ss));
         return 0;
 }
+
+int varlink_server_contains_socket(sd_varlink_server *s, const char *address) {
+        assert(s);
+        assert(address);
+
+        LIST_FOREACH(sockets, ss, s->sockets)
+                if (path_equal(ss->address, address))
```

Comment:
hmm, this might be an abstract namespace socket too, where path_equal() is a bit weird...

maybe add a proper helper here: varlink_server_socket_has_address() which handles this properly, i.e. checks for first char: if @ does an streq() and if / does a path_equal()